src/interfaces/utils/IWethDepositWrapper.sol

```diff
@@ -0,0 +1,70 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.25;
+
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import "../external/lido/IWeth.sol";
+import "../external/lido/ISteth.sol";
+import "../external/lido/IWSteth.sol";
+
+import "../IVault.sol";
+
+/**
+ * @title IWethDepositWrapper
+ * @notice Interface defining the functions for wrapping tokens before deposit into a vault.
+ */
+interface IWethDepositWrapper {
+    /// @dev Errors
+    error AddressZero();
+    error InvalidToken();
+    error InvalidAmount();
+    error InvalidTokenList();
+    error InvalidSender();
+
+    /**
+     * @notice Returns the address of the WETH token.
+     * @return The address of the WETH token.
+     */
+    function weth() external view returns (address);
+
+    /**
+     * @notice Returns the address of the vault to which deposits are made.
+     * @return The address of the vault.
+     */
+    function vault() external view returns (IVault);
+
+    /**
+     * @notice Deposits specified tokens into the vault, converting them to the required format if necessary.
+     * @param to The address that will receive the resulting LP tokens.
+     * @param token The address of the token to deposit (can be WETH, stETH, wstETH, or ETH).
```

Comment:
Description mismatches with implementation and name. There is only WETH an ETH are wrapping.